src/main/java/net/openhft/chronicle/wire/VanillaFieldInfo.java

```diff
@@ -18,111 +18,94 @@
 package net.openhft.chronicle.wire;
 
 import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.core.UnsafeMemory;
 import net.openhft.chronicle.core.util.ObjectUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import java.lang.reflect.Field;
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
 import java.util.Objects;
-import java.util.stream.Collectors;
-
-import static net.openhft.chronicle.wire.WireMarshaller.WIRE_MARSHALLER_CL;
 
 @SuppressWarnings("rawtypes")
 public class VanillaFieldInfo extends AbstractFieldInfo implements FieldInfo {
 
+    private static final long UNSET_OFFSET = Long.MAX_VALUE;
+
     private final Class parent;
     private transient Field field;
+    private transient long offset = UNSET_OFFSET;
 
+    /**
+     * @deprecated This will become package-private in the future, use {@link FieldInfo#createForField(String, Class, BracketType, Field)} instead.
+     */
+    @Deprecated
     public VanillaFieldInfo(String name, Class type, BracketType bracketType, @NotNull Field field) {
         super(type, bracketType, name);
         parent = field.getDeclaringClass();
         this.field = field;
     }
 
+    /**
+     * @deprecated Use {@link FieldInfo#lookupClass(Class)} instead (to be removed in x.25)
+     */
     @NotNull
+    @Deprecated
     public static Wires.FieldInfoPair lookupClass(@NotNull Class aClass) {
-        final SerializationStrategy ss = Wires.CLASS_STRATEGY.get(aClass);
-        switch (ss.bracketType()) {
-            case NONE:
-            case SEQ:
-                return Wires.FieldInfoPair.EMPTY;
-            case MAP:
-                break;
-            default:
-                // assume it could be a map
-                break;
-        }
-
-        @NotNull List<FieldInfo> fields = new ArrayList<>();
-        final WireMarshaller marshaller = WIRE_MARSHALLER_CL.get(aClass);
-        for (@NotNull WireMarshaller.FieldAccess fa : marshaller.fields) {
-            final String name = fa.field.getName();
-            final Class<?> type = fa.field.getType();
-            final SerializationStrategy ss2 = Wires.CLASS_STRATEGY.get(type);
-            final BracketType bracketType = ss2.bracketType();
-            fields.add(new VanillaFieldInfo(name, type, bracketType, fa.field));
-        }
-        return new Wires.FieldInfoPair(
-                Collections.unmodifiableList(fields),
-                fields.stream().collect(Collectors.toMap(FieldInfo::name, f -> f)));
+        return FieldInfo.lookupClass(aClass);
```

Comment:
It seemed like the right thing to do to move this to `FieldInfo` because VanillaFieldInfo is no longer the only implementation.